File: src/resources/templates/basic/java/objects/MyLangString.java
package mylang.objects;

import org.ascore.lang.objects.ASCObject;
import org.ascore.tokens.Token;

/**
 * An example of an object for the MyLang programming main.language
 */
public record MyLangString(String value) implements ASCObject<String> {
    public MyLangString(Token token) {
        this(token.getValue().substring(1, token.getValue().length() - 1)); // removing the enclosing `"` from the string
    }

    @Override
    public String getValue() {
        return value;
    }
}

File: src/resources/templates/basic/kotlin/objects/MyLangString.kt
package mylang.objects

import org.ascore.lang.objects.ASCObject
import org.ascore.tokens.Token

/**
 * An example of an object for the MyLang programming main.language
 */
data class MyLangString(private val value: String) : ASCObject<String> {
    constructor(token: Token) : this(
        token.value.substring(1, token.value.length - 1) // removing the enclosing `"` from the string
    )

    override fun getValue(): String = value
}